pkg/proxy/endpointslicecache.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+import (
+	"sort"
+
+	discovery "k8s.io/api/discovery/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+	utilproxy "k8s.io/kubernetes/pkg/proxy/util"
+	utilnet "k8s.io/utils/net"
+)
+
+// EndpointSliceCache is used as a cache of EndpointSlice information
+type EndpointSliceCache struct {
+	slicesByService  map[types.NamespacedName]map[string]EndpointSliceInfo
+	makeEndpointInfo makeEndpointFunc
```

Comment:
mutateEndpointInfo
document this func, when is this used? is it only for testing?

Reply:
The `makeEndpointInfo` name is used elsewhere in kube-proxy to refer to `makeEndpointFunc`. The whole thing is necessary for a special type used by the iptables proxier here: https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L142. If I were to rename this, should I rename it everywhere else it's called `makeEndpointInfo` within kube-proxy?